mountpoint-s3/src/cli.rs

```diff
@@ -985,27 +1025,135 @@ where
     let fs = S3Filesystem::new(client, prefetcher, bucket_name, prefix, filesystem_config);
     let fuse_fs = S3FuseFilesystem::new(fs);
     tracing::debug!(?fuse_session_config, "creating fuse session");
-    let session = Session::new(fuse_fs, &fuse_session_config.mount_point, &fuse_session_config.options)
-        .context("Failed to create FUSE session")?;
+    let mount_point_path = format!("{}", &fuse_session_config.mount_point);
+    let session = match fuse_session_config.mount_point {
+        MountPoint::Directory(path) => {
+            Session::new(fuse_fs, path, &fuse_session_config.options).context("Failed to create FUSE session")?
+        }
+        #[cfg(target_os = "linux")]
+        MountPoint::FileDescriptor(fd) => Session::from_fd(
+            fuse_fs,
+            fd,
+            session_acl_from_mount_options(&fuse_session_config.options),
```

Comment:
So for ACL there are two related options, `--allow-other` and `--allow-root`. According to https://man7.org/linux/man-pages/man8/mount.fuse3.8.html, `--allow-root` is a libfuse specific option - and it's not passed-to/supported-by kernel - allows only root and filesystem owner to access to the filesystem. `--allow-other` is supported by kernel and allows all users to access to the filesystem. And if you don't pass anything only filesystem owner can access to the filesystem (this is the default behaviour).

On top of what's passed to kernel/libfuse, fuser also implements this logic in userspace https://github.com/awslabs/mountpoint-s3/blob/main/vendor/fuser/src/request.rs#L77-L143.

With FUSE fd mode, one can pass different set of options to kernel and Mountpoint/fuser. But I think it'd still work as expected as long as options passed to any except `--allow-root` as it's not supported by kernel and needs to be passed to Mountpoint.

Reply:
so, for the FUSE fd mode:
1. `--allow-other` must be passed as an argument to the `mount` call and may not be passed to MP as it will be ignored (or not completely ignored according to the [code](https://github.com/awslabs/mountpoint-s3/blob/main/vendor/fuser/src/request.rs#L77-L143)?);
2. `--allow-root` must not be specified via `mount` call and this mode may only be enabled by passing it to MP.

We're [failing](https://github.com/awslabs/mountpoint-s3/pull/1103/files#diff-0448e88502e43140837043cf9b0ac8a46803893dc15863f87f1f6bb290fd88b3R572) now if any of those 2 args is specified, right? Should we allow them? Is it possible to have a test (at least manual) to see that those 2 modes work as expected with fd mode?

Reply:
Addressed with https://github.com/awslabs/mountpoint-s3/pull/1103/commits/30206096434310388b8b5376a952c22738240a7d

Reply:
thanks for fixing this!